jcl/src/openj9.sharedclasses/share/classes/com/ibm/oti/shared/SharedClassStatistics.java

```diff
@@ -45,69 +46,125 @@ public static long maxSizeBytes() {
-     * 
-     * @return the minimum shared classes cache space reserved for AOT data in bytes or -1 if it is not set.
-     */
-    public static long minAotBytes() {
-    	return minAotBytesImpl();
-    }
-    
-    /**
-     * Returns the maximum space allowed for AOT data of the cache that the JVM is currently
-     * connected to.
-     * 
-     * @return the maximum shared classes cache space allowed for AOT data in bytes or -1 if it is not set.
-     */
-    public static long maxAotBytes() {
-    	return maxAotBytesImpl();
-    }
-    
-    /**
-     * Returns the minimum space reserved for JIT data of the cache that the JVM is currently
-     * connected to.
-     * 
-     * @return the minimum shared classes cache space reserved for JIT data in bytes or -1 if it is not set.
-     */
-    public static long minJitDataBytes() {
-    	return minJitDataBytesImpl();
-    }
-    
-    /**
-     * Returns the maximum space allowed for JIT data of the cache that the JVM is currently
-     * connected to.
-     * 
-     * @return the maximum shared classes cache space allowed for JIT data in bytes  or -1 if it is not set.
-     */
-    public static long maxJitDataBytes() {
-    	return maxJitDataBytesImpl();
-    }
+	 * Returns the SysV shmem nattch value for a non-persistent top level cache.
+	 * On Windows or for persistent caches, -1 is returned. Depending on the platform
+	 * and OS, such as z/OS, the value indicates the number of JVMs attached to the cache.
+	 *
+	 * @return the SysV shmem nattch value for a top level non-persistent cache, or -1 if it cannot be determined.
+	 */
+	public static long numberAttached() {
+		return numberAttachedImpl();
+	}
 
+	/**
+	 * Returns the full cache path.
+	 *
+	 * @return the full cache path, or NULL if it cannot be determined.
+	 */
+	public static String cachePath() {
+		return cachePathImpl();
+	}
 	
-	private static native long maxSizeBytesImpl();
+	/**
+	 * Returns the cache name.
+	 *
+	 * @return the cache name, or NULL if it cannot be determined.
+	 */
+	public static String cacheName() {
+		String cachePath = cachePathImpl();
+		if (null != cachePath) {
+			String fileName = new File(cachePath).getName();
+			int prefixIndex = fileName.indexOf('_');
+			int tailIndex = fileName.lastIndexOf('_');
+			if (tailIndex > prefixIndex) {
+				String cacheName = fileName.substring(prefixIndex + 1, tailIndex);
+				if ((prefixIndex > 0) && (fileName.charAt(prefixIndex - 1) != 'P')) {
+					/* Remove "memory_" from the non-persistent cache name. */
+					int typeIndex = cacheName.indexOf('_');
+					if (typeIndex > -1) {
+						cacheName = cacheName.substring(typeIndex + 1);
+					}
+				}
+				return cacheName;
+			}
+		}
+		return null;
+	}
 
+	/**
+	 * Returns the cache dir.
+	 *
+	 * @return the cache dir, or NULL if it cannot be determined.
+	 */
+	public static String cacheDir() {
+		String cachePath = cachePathImpl();
+		if (null != cachePath) {
+			File f = new File(cachePath);
```

Comment:
Please either choose a better name than `f`, or inline into the expression below.